dnSpy/dnSpy/Documents/TreeView/DocumentTreeView.cs

```diff
@@ -844,6 +845,38 @@ sealed class AddDocumentInfo {
 			}
 		}
 
+		static void ResolveInternetShortcutFiles(string[] filenames) {
+			for (var i = 0; i < filenames.Length; i++) {
+				var filename = filenames[i];
+				if (!filename.EndsWith(".url", StringComparison.OrdinalIgnoreCase))
+					continue;
+				if (!File.Exists(filename))
+					continue;
+				try {
+					var urlString = GetUrlFromInternetShortcutFile(filename);
+					if (urlString == null)
+						continue;
+					var urlUri = new Uri(urlString);
```

Comment:
Perhaps instead of using the `Uri` constructor directly, we could use the `Uri.TryCreate` method paired with an `if` statement to avoid throwing an exception on an invalid input string. (Throwing an exception and catching it takes longer than using an `if` statement).

Reply:
Done, I've also removed the try/catch since it really shouldn't throw anything now.